src/media/js/site/login.js

```diff
@@ -14,8 +14,39 @@ export class DefaultUnauthorizedHandler extends React.Component {
 };
 
 
+export class DefaultTOSSignatureHandler extends React.Component {
```

Comment:
Ideally, we could put the TOS stuff in its own file, and have loginRequired + tosRequired be composable.